### PR TITLE
http: move http client creation from NewHttpClientWithInterfaces to N…

### PR DESCRIPTION
**Pull request by ss89**
…ewHttpClient func;

### CODE REVIEW

**Comment by ss89** [comment]
@selm0 could you quickly review this?

File: pkg/http/client.go [hunk 1]
```diff
@@ -76,22 +76,22 @@ func NewHttpClient(config cfg.Config, logger mon.Logger) Client {
...
 	httpClient.SetRedirectPolicy(resty.FlexibleRedirectPolicy(10))
 	httpClient.SetRetryCount(settings.RetryCount)
 	httpClient.SetTimeout(settings.Timeout)
 	httpClient.SetRetryWaitTime(settings.RetryWaitTime)
 	httpClient.SetRetryMaxWaitTime(settings.RetryMaxWaitTime)
 
+	return NewHttpClientWithInterfaces(logger, mo, httpClient)
+}
+
+func NewHttpClientWithInterfaces(logger mon.Logger, mo mon.MetricWriter, httpClient *resty.Client) Client {
```
Comment (selm0) [inline pkg/http/client.go:89]: What's your use case here? `httpClient *resty.Client`: I don't see the benefit of directly passing the client to the test method..

**Review by selm0 — CHANGES_REQUESTED** [review]

File: pkg/http/client.go (hunk 1 above)
Comment (ss89) [reply pkg/http/client.go]: sorry, as said should have been an interface. Was just too sleepy to realize i was passing a pointer

**Comment by ss89** [comment]
@selm0 could you have a quick look and approve?

**Comment by ss89** [comment]
@selm0 ?

**Review by selm0 — APPROVED** [review]

**Pull request merged**